### ISSUE TITLE
LocaleRoute doesn't collect "as" param(s) of parent(s) Route::group

### ISSUE THREAD

**Issue by enstain** [cariboufute/locale-route#4]
## Detailed description

Example case:
```php
Route::group(['as' => 'Backend::'], function() {
    Route::group(['as' => 'Investments.'], function() {
        LocaleRoute::get('checkout', 'InvestmentsController@checkout');
    });
});
```

In such case I expect to call route "Backend::Investments.checkout", but I've got an error
```bash
Route [nl.Backend::Investments.checkout] not defined. 
```

If I try to call "checkout" route, it is "not defined" too. So, I can't put LocaleRoute into Route::group if param "as" is used here, it's just uncallable from app.

## Context

This would be useful to define separate LocaleRoute inside Route::group, because my purpose is to localize GET methods, but keep POST and all other unlocalized

**Comment by cariboufute** [comment]
This would need to implement an equivalent of Illuminate\Routing\Route class in the package. I am thinking seriously about it, but it would need at least a few days to implement this. I keep this issue open though.

**Comment by cariboufute** [comment]
Have you checked your routes with `artisan route:list`? The problem is that you will have the route `Backend::Investments.nl.checkout` instead of `nl.Backend::Investments.checkout`. The good news is that I won't need to make an equivalent of Illuminate Route.

Will check how to fix this...

**Comment by cariboufute** [comment]
For the moment, the workaround would to have a top LocaleRoute::group in your hierarchy for localized routes and subgroups, and normal routes/groups apart.

For your example,

``` php
LocaleRoute::group([], function () {
    Route::group(['as' => 'Backend::'], function() {
        Route::group(['as' => 'Investments.'], function() {
            Route::get('checkout', 'InvestmentsController@checkout');
        });
    });
}
```

I have include a more detailed example in the _readme_ file.

**Comment by enstain** [comment]
As you notice in readme: "If you want to have localized and unlocalized routes in the same group, you will need to duplicate the grouping for both situations.". That's exactly what I want to avoid when trying to keep GET routes localized and POST routes unlocalized, but it seems to be impossible. So, just sad(

Another point is that LocaleRoute::group doesn't pick up translations from routes.php lang file, as I understand. It just adds locale code prefix before uri, but I want to translate all uri parts.

So, working recipe is very WET. I have to define LocaleRoute::get with full path for each route:

``` php
app/Http/routes.php:

Route::group([
  'namespace'   => 'Backend', 
  ...
  ], function() {

      LocaleRoute::get(
        'Backend:Investments.ongoing',
        'InvestmentsController@ongoing'
      );

      LocaleRoute::get(
        'Backend:Investments.agreement',
        'InvestmentsController@agreement'
      );

});
```

And then I define translation of full uri for each route:

``` php
lang/en/routes.php:

'Backend:Investments' => [
    'ongoing' => 'backend-locale/investments-locale/ongoing-locale',
    'agreement' => 'backend-locale/investments-locale/agreement-locale',
],
```

My expectation from route groups localization is that I define the hierarchy of routes like 

``` php
app/Http/routes.php:
['as' => 'foo'], function() {
    ['as' => 'bar'], function() {
       ['as' => 'baz'], function() {}
    }
}
```

and then define translation of each uri part. But it seems impossible too, because routes.php should contain each translation from collection of routes ["foo", "foo.bar", "foo.bar.baz"] to know how construct every localized path "en/foo-locale", "en/foo-locale/bar-locale", "en/foo-locale/bar-locale/baz-locale". One possible way I see to implement such thing is to complicate routes.php lang file structure:

``` php
lang/en/routes.php:

'foo' => [
    'uri' => 'foo-locale',
    'children' => [
        'bar' => [
            'uri' => 'bar-locale',
            'children' => [
                 'baz' => 'baz-locale'
            ]
        ]
    ]
]
```

But, of course, there is also a lot of work with parsing such structure and find required locale uri by route name, so it's your vote whether such thing is important or not.

**Comment by cariboufute** [comment]
@enstain To put unlocalized and localized routes together would need to make a new version of Illuminate\Routing\Router that would treat the 'as' and 'prefix' prefixes differently for locales, so they would be put always first and not at the end like the original Router does.

Since it is more complicated to do, I don't plan it to work on this in the next few days.

For your other point, please put it in another issue so this thread stays on the original subject.

**Comment by cariboufute** [comment]
@enstain Finally, I have done it! It took a lot of hours and crawling into Laravel classes, but I did my own version of Router that decorates the Laravel Router by adding locale to route name and url and by refreshing the route collection.

Using the package is now even easier, and a lot easier. Please check documentation.

So now, you can add *LocaleRoute* methods anywhere in group, and it will manage the group attributes correctly.

Thanks to this improvement, *LocaleRoute::group* method and *SubRoute* class have been removed since they are not necessary anymore.

The changes are in the *develop* branch. They will be merged shortly in *master* with tag **1.0.0-beta5**
